wakuv2/waku.go

```diff
@@ -1206,6 +1208,10 @@ func (w *Waku) Query(ctx context.Context, peerID peer.ID, query store.FilterCrit
 	return result.Cursor(), envelopesCount, nil
 }
 
+func (w *Waku) OnNewEnvelope(env *protocol.Envelope) error {
```

Comment:
Wondering what's the difference with `OnNewEnvelope` and `OnNewEnvelopes`, looks pretty confusing.

Reply:
`OnNewEnvelope` is the new interface which is just a wrapper invoked by underlying FilterManager on receiving any new messages via Filter. 

`OnNewEnvelopes` is an existing method that actually does processing of new envelopes. 

I do see the confusion it causes, let me add some comments to give clarity.